internal/operator-controller/applier/boxcutter.go

```diff
@@ -0,0 +1,225 @@
+package applier
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"hash"
+	"io/fs"
+	"slices"
+	"sort"
+
+	"github.com/davecgh/go-spew/spew"
+	ocv1 "github.com/operator-framework/operator-controller/api/v1"
+	"github.com/operator-framework/operator-controller/internal/operator-controller/controllers"
+	"github.com/operator-framework/operator-controller/internal/operator-controller/rukpak/convert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	revisionHashAnnotation = "olm.operatorframework.io/hash"
+	revisionHistoryLimit   = 5
+)
+
+type Boxcutter struct {
+	Client client.Client
+	Scheme *runtime.Scheme
+}
+
+func (bc *Boxcutter) Apply(
+	ctx context.Context, contentFS fs.FS,
+	ext *ocv1.ClusterExtension,
+	objectLabels, storageLabels map[string]string,
+) ([]client.Object, string, error) {
+	objs, err := bc.apply(ctx, contentFS, ext, objectLabels, storageLabels)
+	return objs, "", err
+}
+
+func (bc *Boxcutter) apply(
+	ctx context.Context, contentFS fs.FS,
+	ext *ocv1.ClusterExtension,
+	objectLabels, _ map[string]string,
+) ([]client.Object, error) {
+	reg, err := convert.ParseFS(contentFS)
+	if err != nil {
+		return nil, err
+	}
+
+	watchNamespace, err := GetWatchNamespace(ext)
+	if err != nil {
+		return nil, err
+	}
+
+	plain, err := convert.PlainConverter.Convert(reg, ext.Spec.Namespace, []string{watchNamespace})
+	if err != nil {
+		return nil, err
+	}
+
+	// objectLabels
+	objs := make([]ocv1.ClusterExtensionRevisionObject, 0, len(plain.Objects))
+	for _, obj := range plain.Objects {
+		labels := obj.GetLabels()
+		if labels == nil {
+			labels = map[string]string{}
+		}
+		for k, v := range objectLabels {
+			labels[k] = v
+		}
```

Comment:
nit: This could be simplified using the new-ish `maps` package in the standard library.

```suggestion
labels := maps.Clone(obj.GetLabels())
maps.Copy(labels, objectLabels)
```

Reply:
TIL